cmd/kube-scheduler/app/server.go

```diff
@@ -238,7 +244,15 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 
 	// Start up the healthz server.
 	if cc.SecureServing != nil {
-		handler := buildHandlerChain(newHealthEndpointsAndMetricsHandler(&cc.ComponentConfig, cc.InformerFactory, isLeader, checks, readyzChecks), cc.Authentication.Authenticator, cc.Authorization.Authorizer)
+		unsecuredMux := newHealthEndpointsAndMetricsHandler(&cc.ComponentConfig, cc.InformerFactory, isLeader, checks, readyzChecks)
+
+		if utilfeature.DefaultFeatureGate.Enabled(zpagesfeatures.ComponentFlagz) {
```

Comment:
Can we move this to the `newHealthEndpointsAndMetricsHandler`, then no need to change the return type. We can rename the function to like `newEndpointsHandler`.

Reply:
Sounds good. Updated.